script.js

```diff
@@ -1,4 +1,110 @@
+var computerhand = ``;
+var winCount = 0;
+var roundCount = 0;
+var gameMode = `waiting for username`;
+var userName = "";
+var playerHand = ``;
+
+// ======================================MAIN FUNCTION STARTS BELOW ====================================
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  myOutputValue = ``;
+  //Username needed to start the game
+  if (gameMode == `waiting for username`) {
+    userName = input;
+    gameMode = `SPS`;
+    myOutputValue = `Hello ${userName}! <br> <br> You are now ready to play. <br> <br> Paper, scissors, or stone?`;
+  } else if (gameMode == `SPS`) {
+    myOutputValue = spsGame(input);
+  }
+
+  return `${myOutputValue} <br> <br> GAME MODE: ${gameMode} <br> <br> Rounds played: ${roundCount}`;
+};
+
+// ===============================MAIN FUNCITON ENDS HERE==================================================
+
+// This function is to compare hands and check whether player or computer won
+var WinnerChecker = function (playerHand, computerhand) {
+  // THIS FUNCTION IS FOR PLAYER WINS
+  if (
+    (playerHand == `scissors` && computerhand == `paper`) ||
+    (playerHand == `paper` && computerhand == `stone`) ||
+    (playerHand == `stone` && computerhand == `scissors`)
+  ) {
+    return `Player win`;
+  }
+
+  // THIS FUCTION IS FOR TIE
+  if (
+    (playerHand == `scissors` && computerhand == `scissors`) ||
+    (playerHand == `paper` && computerhand == `paper`) ||
+    (playerHand == `stone` && computerhand == `stone`)
+  ) {
+    return `Tie`;
+  }
+
+  // THIS IS FOR PLAYER LOSES
+  if (
+    (playerHand == `scissors` && computerhand == `stone`) ||
+    (playerHand == `paper` && computerhand == `scissors`) ||
+    (playerHand == `stone` && computerhand == `paper`)
+  ) {
+    return `Player loses`;
+  }
+};
+
+// SPS Game function
+var spsGame = function (playerHand) {
+  var msg = ``;
+
+  randomnumber = generaterandomnumber();
```

Comment:
you should declare `randomnumber` as a `var`, even thought it may seem to work without, it's not the best practice